sklearn/decomposition/nmf.py

```diff
@@ -55,55 +58,117 @@ def check_non_negative(X, whom):
         raise ValueError("Negative values in data passed to %s" % whom)
 
 
-def _initialize_nmf(X, n_components, variant=None, eps=1e-6,
+def _check_init(A, shape, whom):
+    A = check_array(A)
+    if np.shape(A) != shape:
+        raise ValueError('Array with wrong shape passed to %s. Expected %s, '
+                         'but got %s ' % (whom, shape, np.shape(A)))
+    check_non_negative(A, whom)
+    if np.max(A) == 0:
```

Comment:
I think this is not the right way to do it. In this case, the optimal solution is probably to not update the coefficient.